daemon/src/main/java/org/lsposed/lspd/cli/handler/ScopeHandler.java

```diff
@@ -0,0 +1,228 @@
+package org.lsposed.lspd.cli.handler;
+
+import android.os.RemoteException;
+
+import com.beust.jcommander.JCommander;
+import com.beust.jcommander.Parameter;
+import com.beust.jcommander.Parameters;
+import com.beust.jcommander.ParameterException;
+
+import org.lsposed.lspd.ILSPManagerService;
+import org.lsposed.lspd.cli.Utils;
+import org.lsposed.lspd.cli.exception.FormatException;
+import org.lsposed.lspd.cli.exception.NoEnoughArgumentException;
+import org.lsposed.lspd.cli.exception.NotXposedModuleException;
+import org.lsposed.lspd.cli.exception.RunException;
+import org.lsposed.lspd.cli.exception.UninstalledPackageException;
+import org.lsposed.lspd.cli.exception.UnknownCommandException;
+import org.lsposed.lspd.models.Application;
+
+import static org.lsposed.lspd.cli.Utils.ERRCODES;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+
+import io.github.xposed.xposedservice.utils.ParceledListSlice;
+
+@Parameters(
+    commandNames = { "scope" },
+    commandDescription = "Manage scope for module"
+)
+public class ScopeHandler implements ICommand {
+    private LSCommand lsCommand;
+    private AppendCommand appendCommand;
+    private SetCommand setCommand;
+    private RemoveCommand removeCommand;
+    private JCommander jc;
+
+    @Parameters(commandNames = "ls", commandDescription = "List module scope")
+    public static class LSCommand implements ICommand {
+        @Parameter(description = "<Module package name>", required = true)
+        public String packageName;
+
+        public void prepareSubCommands(JCommander jc) {}
+
+        public int run(ILSPManagerService manager) throws RemoteException, NoEnoughArgumentException, NotXposedModuleException {
+            if (!Utils.validXposedModule(manager, packageName)) {
+                throw new NotXposedModuleException(packageName);
+            }
+
+            var scope = manager.getModuleScope(packageName).getList();
+            for (var app : scope) {
+                System.out.println(app.packageName + "/" + app.userId);
+            }
+            return ERRCODES.NOERROR.ordinal();
+        }
+    }
+
+    public static class ScopeEditCommand implements ICommand {
+        @Parameter(names = {"-m", "--module"}, description = "<Module package name>", required = true, order = 0)
+        public String packageName;
+        @Parameter(description = "scope")
+        public List<String> scope;
+        @Parameter(names = {"-i", "--ignore"}, description = "Will not throw an exception if app not installed")
+        public boolean ignore = false;
+        protected List<Application> apps;
+
+        public void prepareSubCommands(JCommander jc) {}
+
+        public int run(ILSPManagerService manager) throws NoEnoughArgumentException, FormatException, RemoteException, UninstalledPackageException, NotXposedModuleException, RunException {
+            boolean bMsgReboot = false;
+
+            if (packageName == null) {
```

Comment:
Does it really happen? I think JCommander will throw `com.beust.jcommander.ParameterException`

Reply:
I removed, it was useless check